File: pset6/readability/readability.py
import re
from cs50 import get_string

text = get_string("Text: ")

words = len(text.split())

sentences = len(re.split("\. |\? |\! ", text))

count = 0
for i in text:
    if i.isalpha() == True:
        count = count + 1
letters = count


grade = round(0.0588 * ((letters/words)*100) - 0.296 * ((sentences/words)*100) - 15.8)

if grade < 1:
    print("Before Grade 1")
elif grade > 15:
    print("Grade 16+")
else:
    print(f"Grade {grade}")

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
#include <string.h>
#include <ctype.h>

int main(void)
{

 float letters;
 float words;
 float space = 0;
 float upper = 0;

string Text = get_string("Text:\n");

int x = strlen(Text);
    for (int i = 0, n = strlen(Text); i < n; i++)
    {

        if isalpha(Text[i])
        {
            letters = letters + 1;
        }

        if isblank(Text[i])
        {
            space = space + 1;
        }
        if (Text[i]=='.' || Text[i]=='!' || Text[i]=='?')
        {
            upper = upper + 1;
        }
    }

words = space + 1;

//printf("%f letter(s)\n", letters);
//printf("%f word(s)\n", words);
//printf("%f sentence(s)\n", upper);

float L = 100*letters/words;
float S = 100*upper/words;

float index = 0.0588 * L - 0.296 * S - 15.8;
int grade = rint(index);

   if (index < 1)
      {
         printf("Before Grade 1\n");
      }
   if (index >= 16)
      {
         printf("Grade 16+\n");
      }
   if (index >= 1 && index < 16)
      {
         printf("Grade %i\n", grade);
      }
}